src/pkcs15init/pkcs15-lib.c

```diff
@@ -2039,12 +2041,18 @@ check_key_compatibility(struct sc_pkcs15_card *p15card, struct sc_pkcs15_prkey *
 			}
 		}
 		else if (key->algorithm == SC_ALGORITHM_EC)   {
+			if (!sc_valid_oid(&key->u.ec.params.id))
+				if (sc_pkcs15_fix_ec_parameters(ctx, &key->u.ec.params))
+					LOG_FUNC_RETURN(ctx, SC_ERROR_OBJECT_NOT_VALID);
+			if (sc_valid_oid(&info->u._ec.curve_oid))
```

Comment:
Is it possible to use custom domain parameters, which don't have an OID. I know a card that can be configured that way, however, in practice it is not used.